### DISCUSSION TITLE
Eurostile bold extended theme

### DISCUSSION THREAD

**Discussion by ghost** [GitSquared/edex-ui#1164] [Ideas]
Eurostile bold extended was used by 2001: A Space Odyssey and has been used in lots of movies since to indicate "THE FUTURE":
https://typesetinthefuture.com/2014/11/29/fontspots-eurostile/
It would be cool if there was a default theme using this font.  Even cooler if we could get a HAL-9000 component :D

**Comment by GitSquared** [comment]
Funny, I was just reading Typeset In The Future yesterday and thinking about this.

eDEX was orginally built as a "working copy" of [DEX](https://github.com/seenaburns/dex-ui), so I pretty much copied everything without asking questions at first (including the font - which admittedly looks much like the [scene](https://gmunk.com/TRON-Board-Room) this is all inspired from).

I don't think I would support a change of the default font in eDEX, however there are [ideas floating around](https://github.com/GitSquared/edex-ui/discussions/1028#discussioncomment-354827) of a spiritual successor where we could take much more creative freedom.

In the meantime would love to see a theme with  Eurostile - I wonder if there are licensing issues though?

**Reply by ghost** [reply]
Sorry, by "default theme", I just meant one that shipped with the software, not one that a user would have to come up with themselves.  The tron theme should of course be the standard one.

**Reply by ghost** [reply]
As far as licensing, Eurostile is among the Google fonts:
https://fonts.google.com/?query=eurostile
and there's also Michroma
https://fonts.google.com/specimen/Michroma#standard-styles

**Reply by GitSquared** [reply]
It's listed in Google Fonts, but not under a permissive license:
![Screenshot_20210815-221916_Brave.jpg](https://user-images.githubusercontent.com/24496417/129491613-51e6b114-4ea4-4396-89a3-e13f90761ff7.jpg)

**Reply by ghost** [reply]
Ok. I know Michroma is FOSS.

**Comment by ghost** [comment]
Michroma is a particularly wide font. It ends up looking like the font size is too big, as well as having some overlap issues in the clock:
![Screen Shot 2021-08-16 at 2 37 08 PM](https://user-images.githubusercontent.com/30355933/129627118-ae790786-af4c-4f8b-ace4-600246c0ce6f.png)
What do you think the best way to fix that would be?  Add a variable that lets you shrink the font size? Compute the font size from the width of some glyphs?

**Reply by GitSquared** [reply]
> Add a variable that lets you shrink the font size?

I would go for that.

**Reply by ghost** [reply]
Maybe something like this would work better:
https://css-tricks.com/fun-viewport-units/
With vmin, vmax, and calc, one could get text that is as tall as possible without being too wide.
